gpu/kinfu_large_scale/tools/kinfuLS_app.cpp

```diff
@@ -977,29 +981,59 @@ struct KinFuLSApp
   }
 
   void
-  startMainLoop (bool triggered_capture)
+  startMainLoop (bool triggered_capture, bool oni2_interface = false)
   {   
-    using namespace openni_wrapper;
-    using DepthImagePtr = boost::shared_ptr<DepthImage>;
-    using ImagePtr = boost::shared_ptr<Image>;
+    boost::signals2::connection c;
 
-    std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
-    {
-      source_cb2 (img, depth, constant);
-    };
-    std::function<void (const DepthImagePtr&)> func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
-    std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = [this] (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& cloud)
+    if(oni2_interface)
     {
-      source_cb3 (cloud);
-    };
+      using namespace pcl::io;
+      using DepthImagePtr = boost::shared_ptr<DepthImage>;
+      using ImagePtr = boost::shared_ptr<Image>;
+
+      std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
+      {
+        source_cb2 (img, depth, constant);
+      };
+      std::function<void (const DepthImagePtr&)> func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
+      std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = [this] (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& cloud)
+      {
+        source_cb3 (cloud);
+      };
+
+      bool need_colors = integrate_colors_ || registration_ || enable_texture_extraction_;
 
-    bool need_colors = integrate_colors_ || registration_ || enable_texture_extraction_;
+      if ( pcd_source_ && !capture_.providesCallback<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&)>() ) {
+        std::cout << "grabber doesn't provide pcl::PointCloud<pcl::PointXYZRGBA> callback !\n";
+      }
 
-    if ( pcd_source_ && !capture_.providesCallback<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&)>() ) {
-      std::cout << "grabber doesn't provide pcl::PointCloud<pcl::PointXYZRGBA> callback !\n";
+      c = pcd_source_? capture_.registerCallback (func3) : need_colors ? capture_.registerCallback (func1) : capture_.registerCallback (func2);
```

Comment:
Maybe use a lambda with if else syntax to still make it understandable

Like so
```
c = [this](const auto xyzrgba_cb, auto& rgbd_cb, some_type default_cb){ 
    if (pcd_source) { return capture_.registerCallback(xyzrgba_cb); }
    if (need_colors) { return capture_.registerCallback(rgbd_cb); }
    return capture_.registerCallback(default_cb); }(func3, func2, func1);
```
```

Reply:
I agree that the ternary statement is not well readable here but your proposal based on lambda seems overkilling to me

I'll propose a classic if/else statement instead